### PR TITLE
Spectator - Update the core meters pages, adding the percentile-timer…

### PR DESCRIPTION
**Pull request by manolama**
… and

linking to the pattern doc.

### CODE REVIEW

File: docs/spectator/core/meters/timer.md [hunk 1]
```diff
@@ -3,8 +3,9 @@
 A Timer is used to measure how long (in seconds) some event is taking. Two types of Timers
 are supported:
 
-* `Timer`: for frequent short-duration events.
-* `LongTaskTimer`: for long-running tasks.
+* `Timer` - for frequent short-duration events.
+* `LongTaskTimer` - for long-running tasks.
+* [Percentile Timer] - for tracking the frequency of measurements.
```
Comment (brharrington) [inline docs/spectator/core/meters/timer.md:8]: "tracking frequency" sounds more like a counter. I think something like "if percentile approximations are needed in addition to basic stats".

Also, percentiles are an extension to a short-duration timer. For this page I think it is probably better to focus on the short and long use-cases. The short duration section can discuss basic timers and then link off to bucket and percentile extensions. Querying is also different between short/long timers.

File: docs/spectator/core/meters/timer.md [hunk 2]
```diff
@@ -29,6 +30,127 @@ Long Task Timer:
...
+    including the `totalTime`, `count`, `max` and `totalOfSquares`. If you were to simply query for
+    the name of your timer via
+
+    ```
+    nf.cluster,foo,:eq,name,http.req.latency,:eq,:and
+    ```
+
+    you would get a nonsense value that is the sum of the reported statistics.
+
+When querying the results of a timer, use one of the ASL functions below to generate
```
Comment (brharrington) [inline docs/spectator/core/meters/timer.md:47]: "ASL" might not be familiar to a new user that comes directly here. I think it is better to just say operator or function.

File: docs/spectator/core/meters/timer.md [hunk 3]
```diff
@@ -29,6 +30,127 @@ Long Task Timer:
...
+a useful response.
+
+### Average Measurement (:dist-avg)
```
Comment (brharrington) [inline docs/spectator/core/meters/timer.md:50]: I think this goes into too much detail that is just going to make it more confusing for a new user just trying to use a timer. The finer details of how the average is computed should probably be placed in the reference page for the `:dist-avg` operator. For this section it seems like enough to just advise using `:dist-avg` and being clear it is an average per recorded value, e.g. request.

File: docs/spectator/core/meters/timer.md [hunk 4]
```diff
@@ -29,6 +30,127 @@ Long Task Timer:
...
+
+For [Timer] and Distribution Summary metrics, the `totalTime` (timers) /`totalAmount` (distributions)
+and `count` are collected each time a measurement is taken. If this technique was applied to a
+request latency metric, then you would have the average latency per request for an arbitrary
+grouping. These types of metrics have an explicit count based on activity. To get an average
+per measurement manually:
+
+```
```
Comment (brharrington) [inline docs/spectator/core/meters/timer.md:58]: Using the `@@@ atlas-stacklang` directive is preferable for these sort of snippets so all operators are linked back to the reference in the example.

File: docs/spectator/core/meters/timer.md [hunk 5]
```diff
@@ -29,6 +30,127 @@ Long Task Timer:
...
+nf.cluster,foo,:eq,
+name,http.req.latency,:eq,:and,
+:cq
+```
+
+This is much simpler using the [:dist-stddev] function:
+
+```
+nf.cluster,foo,:eq,name,http.req.latency,:eq,:and,
+:dist-max
```
Comment (brharrington) [inline docs/spectator/core/meters/timer.md:143]: `:dist-stddev`

File: docs/spectator/core/meters/percentile-timer.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# Percentile Timers
```
[the remark this answers is not in the document]
Comment (manolama) [reply docs/spectator/core/meters/percentile-timer.md:1]: What's the difference between patterns and meters? Each language had a percentile-timer page under the respective meters section so I figured a summary would fit here.

Would it be better to just have a single meters page with descriptions of the main meters, maybe a grid with the links to the proper language implementation?

File: docs/spectator/core/meters/counter.md [hunk 7]
```diff
@@ -1,13 +1,19 @@
 # Counter
 
-A Counter is used to measure the rate at which some event is occurring. Considering a simple 
-queue, Counters could be used to measure things like the rate at which items are being inserted
-and removed.
+A Counter is used to measure the rate at which some event is occurring. A common use is 
+tracking requests to a REST endpoint. Each request to the endpoint should increment, by one, 
+a counter tagged with the name of the endpoint.
```
[the remark this answers is not in the document]
Comment (manolama) [reply docs/spectator/core/meters/counter.md:5]: Ack, I'll restore.

File: docs/spectator/core/meters/percentile-timer.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+!!! Warning
+    Please be selective about what you measure as there is significant overhead on both
+    the client and storage side. Usually only one or two key performance indicators (KPIs)
+    per application. Limit the tag cardinality as much as possible. For example, only
+    include an endpoint tag, not a user agent or response code. Use one of the other meter
```
[the remark this answers is not in the document]
Comment (manolama) [reply docs/spectator/core/meters/percentile-timer.md:7]: I suppose if an instance only has one API endpoint that's fine, but for most API servers you'd definitely add the endpoint so you can differentiate requests to `/api/v1/graph` from `/api/v1/tags`. Adding response code is usually ok but it's also possible to simply track those in a counter and avoid the extra percentile overhead.

What should users tag percentiles with?

File: docs/spectator/core/meters/timer.md (hunk 1 above)
Comment (manolama) [reply docs/spectator/core/meters/timer.md:8]: Ack. Folks reading this page will expect some mention of percentiles, hence the link out. I could drop it at the bottom and tell them to take a look if these don't fit their use case.

Do you have an example of the long timer query patterns?

File: docs/spectator/core/meters/timer.md (hunk 3 above)
Comment (manolama) [reply docs/spectator/core/meters/timer.md:50]: I was kinda wondering why all of that detail was here. I'll move them.

**Review by brharrington — APPROVED** [review]

**Pull request merged**